### PR TITLE
5 get all articles

### PR DESCRIPTION
**Pull request by skelbon**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -74,4 +75,78 @@ describe('GET /api', ()=>{
             expect(hasAllProperties).toBe(true)
         })
     })
+})
+
+describe('GET /api/articles/:article_id', ()=>{
```
Comment (Augs0) [inline __tests__/app.test.js:80]: This looks like code from a different endpoint

File: __tests__/app.test.js [hunk 2]
```diff
@@ -74,4 +75,78 @@ describe('GET /api', ()=>{
...
+        .get('/api/articles/9999')
+        .expect(400)
+        .then(({body})=>{
+            expect(body).toEqual({ message: 'Invalid article_id - the article does not exist or your article_id is malformed' })
+        })
+    })
+})
+
+describe('GET /api/articles', ()=>{
+    test('should return status code 200', ()=>{
```
Comment (Augs0) [inline __tests__/app.test.js:116]: Nice testing !

File: app.model.js [hunk 3]
```diff
@@ -7,3 +7,24 @@ exports.fetchTopics = ()=>{
...
+exports.fetchArticleById = (article_id)=>{
+    return db.query(`SELECT * FROM articles WHERE article_id=$1`, [article_id]).then(({rows})=>{
+        return rows[0] ?? Promise.reject() 
+    }).catch((err)=>{
+        return Promise.reject({ message: 'Invalid article_id - the article does not exist or your article_id is malformed' })
+    })
+}
+
+exports.fetchAllArticles = ()=>{
+    return db.query(`SELECT CAST(COUNT(comments) AS INT) AS
```
Comment (Augs0) [inline app.model.js:19]: Hey, not sure about this line. Can you explain what you're doing here? I'm pretty sure the count should be for comment_id and not the whole table

**Review by Augs0 — CHANGES_REQUESTED** [review]
Overall, looking pretty good but I'm unsure about your SQL query. Could you take a look at it?

File: app.model.js (hunk 3 above)
Comment (skelbon) [reply app.model.js:19]: Hi August. The query is good. The GROUP BY parameter ensures the count is applied to each grouping. It is then applied to the comment_count column.
[playground.txt](https://github.com/skelbon/news-api/files/12800835/playground.txt)

**Pull request merged**